tools/setup/generate_zulip_bots_static_files.py

```diff
@@ -44,6 +50,10 @@ def copyfiles(paths: List[str]) -> None:
     docs = glob.glob(doc_glob_pattern)
     copyfiles(docs)
 
+    assets_glob_pattern = os.path.join(package_bots_dir, "*/assets/*")
+    assets = glob.glob(assets_glob_pattern)
+    copyfiles(assets, are_assets=True)
```

Comment:
This whole parallel pattern is a smell -- instead of iterating over _one_ thing inside the bot dir we want, and then _another_ one thing, and then _another_ one thing, we should iterate over each subdirectory in `get_bots_directory_path()`, and copies (recursively) the set of things we want out.  It should early-abort if the subdirectory doesn't have a `doc.md`.

This also makes it easier to have one place to check if the bot name is in `BOT_INTEGRATIONS` from `zerver.lib.integrations` -- there are only two that we care about, and we're copying over 35 of them.  Skipping the other 33 seems like it would make sense.

Reply:
I have created `copy_bots_data` which copies the required files and folders to the `static/generated/bots` directory. The input of that function should be the `BOT_INTEGRATIONS` present in `zerver.lib.integrations` but I am not able to figure out how to access that from the current file.

I have tried something but it is failing the CI -- it's working on my local machine -- I am not sure how to proceed on this one.

Reply:
Hmm, yeah, for import timing reasons it is probably hard to get at BOT_INTEGRATIONS during provisioning, so doing all of them is probably the simplest plan.